src/Orleans/Streams/Internal/IStreamGrainExtensions.cs

```diff
@@ -11,6 +11,7 @@ namespace Orleans.Streams
     internal interface IStreamConsumerExtension : IGrainExtension
     {
         Task<StreamHandshakeToken> DeliverItem(GuidId subscriptionId, Immutable<object> item, StreamSequenceToken currentToken, StreamHandshakeToken handshakeToken);
```

Comment:
Try giving different names to those 2 method: DeliverMutable and DeliveeImmutable.